### PR TITLE
add setup-r-dependencies

### PR DESCRIPTION
**Pull request by donyunardi**
Related to https://github.com/insightsengineering/nestdevs-tasks/issues/65
Switched to setup-r-dependencies and added these packages for lookup-refs:
* teal.code
* teal.data
* teal.logger
* teal.widgets
* formatters
* rtables

### CODE REVIEW

File: .github/workflows/check.yaml [hunk 1]
```diff
@@ -40,6 +40,14 @@ jobs:
...
       unit-test-report-brand: >-
         https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/thumbs/teal.png
+      deps-installation-method: setup-r-dependencies
+      lookup-refs: |
+        insightsengineering/teal.code
+        insightsengineering/teal.data
+        insightsengineering/teal.logger
+        insightsengineering/teal.widgets
+        insightsengineering/formatters
+        insightsengineering/rtables
```
Comment (m7pr) [inline .github/workflows/check.yaml:50]: @donyunardi why do we need formatters and rtables?

Comment (m7pr) [reply .github/workflows/check.yaml:50]: do we also need `insightsengineering/rtables.officer`?

Comment (m7pr) [reply .github/workflows/check.yaml:50]: Ok it's teal.transform using teal.widgets. teal.widgets is dependent on rtables, and rtables are dependent on formatters

Comment (donyunardi) [reply .github/workflows/check.yaml:50]: You got it!

**Pull request merged**